### DISCUSSION TITLE
How to combine a generated notifier provider with another provider

### DISCUSSION THREAD

**Discussion by snapsl** [rrousselGit/riverpod#2071] [Q&A] [answered]
Hi,

Is there a way to combine a generated provider based on the notifier class with an other provider?

```dart
@riverpod
class Counter extends _$Counter {
  @override
  int build() {
    return 0;
  }

  void increment() {
    state++;
  }
}
```
I would want the generated ``counterProvider`` to watch a ``secondProvider`` like the following:

```dart
final counterProvider = StateNotifierProvider<Counter, int>((ref) {
  ref.watch(secondProvider);
  return Counter(ref);
});
```

Is this possible to do?

**Comment by rrousselGit** [comment, marked as answer]
You can call `ref.watch` inside `build`

**Reply by snapsl** [reply]
@rrousselGit, thank you for the answer. That's exactly what I was looking for, but using ``ref.watch`` inside another function like ``increment()`` does not trigger an update?

Is this an option:
```
@override
int build() {
  ref.listen<type>(
      otherProvider,
      (previous, next) {
        increment();
      },
    );
}
```
